src/BannerWithLink/index.tsx

```diff
@@ -28,75 +26,12 @@ export const BannerWithLink: React.FC<Props> = (props: Props) => {
         return null;
     }
 
-    const {
-        brazeMessageProps: {
-            ophanComponentId,
-            header,
-            body,
-            boldText,
-            buttonText,
-            buttonUrl,
-            imageUrl,
-        },
-        trackClick,
-    } = props;
-
-    const styles = selfServeStyles(props.brazeMessageProps, defaultColors);
+    const { boldText } = props.brazeMessageProps;
 
-    const [showBanner, setShowBanner] = useState(true);
-
-    if (!showBanner) {
-        return null;
+    // StyleableBannerWithLink uses `highlight` instead of `boldText` attribute - this check fixes the discrepency so Marketing won't have to change existing canvases
+    if (boldText) {
+        props.brazeMessageProps.highlight = boldText;
```

Comment:
I wonder if rather than mutating `props.brazeMessageProps`, this could be resolved in the call to `StyleableBannerWithLink` below? That feels slightly cleaner to me (in a general feeling of mutation being best avoided where possible). I.e. instead of spreading the props with `...props` specify them explicitly.

Reply:
Updated